src/java.base/share/native/libjli/java.c

```diff
@@ -455,6 +456,14 @@ JavaMain(void* _args)
         LEAVE();
     }
 
+    /*
+     * -Xshare:dump does not have a main class so the VM can safely exit now
+     */
+    if (dumpSharedSpaces) {
+      CHECK_EXCEPTION_LEAVE(0);
+      LEAVE();
```

Comment:
What is exit status ($?) when -Xshare:dump fails. It looks like any pending exception will be printed and it will exit with 0 but maybe I've missed something.

In passing, the java launcher uses 4-space indent rather than 2.

Reply:
I am not incredibly familiar with the java launcher but I believe what I have is a mistake. I will change it to `CHECK_EXCEPTION_LEAVE(1)` instead.

Reply:
Yes, if you change it CHECK_EXCEPTION_LEAVE(1)then any pending exception will be printed and it will exit with 1; if there is no pending exception then it exit with the value of `ret` as the exit status, which will be 0 here.

Reply:
Sorry I missed this. Unclear when `CHECK_EXCEPTION_LEAVE(0)` would ever be appropriate???

Reply:
I can't think of any case where `CHECK_EXCEPTION_LEAVE(0)` would be useful, in which case we need to check the -version/-showversion handling in the unlikely event that VersionProps.print completes with an exception (nothing to do this PR of course).